src/modules/elixir.rs

```diff
@@ -0,0 +1,74 @@
+use ansi_term::Color;
+use std::process::Command;
+
+use super::{Context, Module};
+
+/// Creates a module with the current Elixir version
+///
+/// Will display the Elixir version if any of the following criteria are met:
+///     - Current directory contains a `.ex` file
+///     - Current directory contains a `mix.exs` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_ex_project = context
+        .try_begin_scan()?
+        .set_files(&["mix.exs"])
+        .set_extensions(&["ex", "exs"])
+        .is_match();
+
+    if !is_ex_project {
+        return None;
+    }
+
+    match get_elixir_version() {
+        Some(elixir_version) => {
+            const ELIXIR_CHAR: &str = "🧪 ";
+
+            let mut module = context.new_module("elixir");
+            let module_style = module
+                .config_value_style("style")
+                .unwrap_or_else(|| Color::Purple.bold());
+            module.set_style(module_style);
+
+            let formatted_version = format_elixir_version(&elixir_version)?;
+            module.new_segment("symbol", ELIXIR_CHAR);
+            module.new_segment("version", &formatted_version);
+
+            Some(module)
+        }
+        None => None,
+    }
+}
+
+fn get_elixir_version() -> Option<String> {
+    match Command::new("elixir").arg("-v").output() {
+        Ok(output) => Some(String::from_utf8(output.stdout).unwrap()),
+        Err(_) => None,
+    }
+}
+
+fn format_elixir_version(version_str: &str) -> Option<String> {
+    let version = version_str
+        // ignore detailed Erlang output
+        .lines()
+        .last()?
+        .trim()
+        // split into ["Elixir", "1.9.1", "(compiled", "with, "Erlang/OTP", "21)"]
+        .split_whitespace()
+        .nth(1)?;
+
+    let otp_version = version_str
+        // ignore detailed Erlang output
+        .lines()
```

Comment:
Can we extract this out so we don't have to evaluate it twice:

```
        .lines()
        .last()?
        .trim()
        .split_whitespace()
```

Reply:
I tried to do that yesterday and got a "cannot borrow version_arr as mutable" error. I'm afraid I don't know how to do that at this point. I looked at the docs for `std::str::SplitWhitespace<'_>` but it wasn't much help.

Reply:
Oh, right. `SplitWhitespace` is an iterator, so you aren't supposed to use it from two places.

There are two obvious solutions:
1. Stop at the `trim()` when extracting out the common code, and repeat the `split_whitepace()` for each version item.
2. `collect()` after the `split_whitespace()` to materialize in to `Vec<&str>` which can safely be used in multiple places.

Option 1 is _probably_ more performant, because it doesn't require any allocation, and the strings are quite small.